### PR TITLE
Fix InputEvent handling order in ViewportContainer

### PR DESCRIPTION
**Pull request by logicor-cn**
See #25525 

UI components inside a contained `Viewport` are expected to follow
the same global InputEvent handling order, which means `_input`
function in any `Node` shall be honored first, then `_gui_input`
function of those under the cursor / focus.

However, currently implementation of `ViewportContainer` uses its
`_input` to capture events for both `_input` and `_gui_input` of
UI `Control` under contained `Viewport`.  Effectively giving their
`_gui_input` the priority of `_input`.  This could:

1. Errornously triggle `_gui_input` UI `Control` inside contained
 `Viewport`
2. Errornously block `_gui_input` UI `Control` from receiving due
 event. (Due to event being consumed by 1)

Separate the event delivery between `ViewportContainer` and `Viewport`
into `_input` and `_gui_input` so proper InputEvent handling order
is restored.

### CODE REVIEW

**Comment by SASUPERNOVA** [comment]
In viewport_container.cpp you are xforming the event using `p_event->xformed_by(xform.affine_inverse());`. This works as intended if the ViewportContainer is positioned at the origin (Vector2(0,0) of the Scene). However, it does not seem to work properly if the ViewportContainer is in a different position. For instance, if the ViewportContainer rect_position is Vector2(10, 10), and it contains a Button with rect_position Vector2(15, 15), the Button will start capturing events in Vector2(25, 25) relative to the origin of the ViewportContainer, instead of the origin of the Scene. I found that getting a "local transform" using `p_event->xformed_by(xform.affine_inverse() * xform)` instead solves this issue.

**Comment by logicor-cn** [comment]
@SASUPERNOVA I am not sure not sure the issue you are talking about but sounds like a different issue than #25525.  I could reproduce #25525 with `ViewportContainer` placed at origin.

**Comment by SASUPERNOVA** [comment]
Yes, it is a different issue than #25525; however, it is an issue relevant to the pull request. I have created a small sample project, to provide a better example of what I meant ([ViewportContainer.zip](https://github.com/godotengine/godot/files/2826311/ViewportContainer.zip)). 

The Scene contains only a ViewportContainer, the Viewport of which contains only a button. Using the code from your pull request, clicking on the button does not emit its pressed() signal. The pressed() signal is only emitted when the cursor is in the area Vector2(70, 70) (upper left corner) to Vector2(100, 100) (bottom right corner). This happens because the rect_position of ViewportContainer is Vector2(30, 30) which is the size of the Button. 

In other words, as it stands now, events of Controls inside a ViewportContainer are triggered relative to the ViewportContainers' position in the Scene. 

However, if `p_event->xformed_by(xform.affine_inverse());` is replaced with `p_event->xformed_by(xform.affine_inverse() * xform);`, events get triggered relative to the Controls' position inside the ViewportContainer, effectively solving the issue.

**Comment by logicor-cn** [comment]
@SASUPERNOVA You are right. I have added suggested change.

**Comment by akien-mga** [comment]
Sorry for the late review. PRs should be made against the `master` branch, not stable branches like `3.0` and `3.1`, so this cannot be merged.

Could you make a new PR against the master branch?

**Comment by akien-mga** [comment]
Closing as this is made against the wrong branch and cannot be merged.

**Comment by fire** [comment]
I made a port to 3.x https://github.com/fire/godot/tree/input-event-handling-order because Sunset Moth from discord asked.
